### PR TITLE
Migrate management-type develop pages from wiki 

### PR DESCRIPTION
**Pull request by martyngigg**
Description of work.

Moves over several management-type pages to the developer rst docs.

**To test:**

* Build the developer docs and check out the new pages.

Refs #22082

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/PatchReleaseChecklist.rst [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+###########
+
+The patch release will be prepared based off the branch used to
+construct to most recent major point release, e.g. ``release-v3.9``
+would be used for any ``3.9.x`` patches.
+
+* changes for the patch should be made using the standard GitHub
+  workflow for merging code with ``master``. The issue and pull request
+  should then have the ``PatchCandidate`` label applied to them. These
+* commits will then be cherry picked from ``master`` on to the release
```
Comment (gvardany) [inline dev-docs/source/PatchReleaseChecklist.rst]: This sentence is broken with *

File: dev-docs/source/JenkinsConfiguration.rst [hunk 2]
```diff
@@ -0,0 +1,459 @@
+.. _JenkinsConfigurationNotes:
```
Comment (gvardany) [inline dev-docs/source/JenkinsConfiguration.rst]: Should this not match the file name?

File: dev-docs/source/JenkinsConfiguration.rst [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+######################
+
+Machine Setup
+-------------
+
+Set up a local ``builder`` account that will be used by the slave.
+
+Install the :ref:`required prerequisites <GettingStartedWithMantid>` for the relevant OS.
+
+For Windows note that the `Command line Visual C++ build tools <http://landinghub.visualstudio.com/visual-cpp-build-tool>`__
```
Comment (gvardany) [inline dev-docs/source/JenkinsConfiguration.rst]: This link seems to be broken now.

**Review by gvardany — COMMENTED** [review]
The pages look good, just few minor problems inline:

File: dev-docs/source/JenkinsConfiguration.rst (hunk 2 above)
Comment (martyngigg) [reply dev-docs/source/JenkinsConfiguration.rst]: Yes it should - good catch.

**Comment by martyngigg** [comment]
Thanks, @gvardany. I've fixed those issues now.

**Comment by gvardany** [comment]
Thanks, looks good :shipit:

**Comment by martyngigg** [comment]
Thanks. Could you mark it as approved on the review section too?

**Review by gvardany — APPROVED** [review]
I couldn't change my earlier review probably because of force push.

**Pull request merged**